flink-connector-aws/flink-connector-aws-kinesis-streams/src/test/java/org/apache/flink/connector/kinesis/source/metrics/KinesisSourceShardMetricsTest.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.kinesis.source.metrics;
+
+import org.apache.flink.connector.kinesis.source.split.KinesisShardSplit;
+import org.apache.flink.connector.kinesis.source.split.KinesisShardSplitState;
+import org.apache.flink.connector.kinesis.source.util.TestUtil;
+import org.apache.flink.metrics.testutils.MetricListener;
+import org.apache.flink.runtime.metrics.groups.AbstractMetricGroup;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.arns.Arn;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatNoException;
+
+class KinesisSourceShardMetricsTest {
+    private static final KinesisShardSplit TEST_SPLIT =
+            TestUtil.getTestSplit(TestUtil.STREAM_ARN, TestUtil.generateShardId(1));
+    private static final KinesisShardSplitState TEST_SPLIT_STATE =
+            TestUtil.getTestSplitState(TEST_SPLIT);
+
+    private MetricListener metricListener;
+    private KinesisSourceShardMetrics kinesisSourceShardMetrics;
+
+    @BeforeEach
+    public void init() {
+        metricListener = new MetricListener();
+
+        kinesisSourceShardMetrics =
+                new KinesisSourceShardMetrics(
+                        metricListener
+                                .getMetricGroup()
+                                .addGroup(MetricConstants.KINESIS_STREAM_SOURCE_METRIC_GROUP));
+
+        kinesisSourceShardMetrics.registerShardMetricGroup(TEST_SPLIT);
+    }
+
+    @Test
+    void testRegisterShardMetricGroup() {
+        TestUtil.assertMillisBehindLatest(TEST_SPLIT, -1L, metricListener);
+    }
+
+    @Test
+    void testKinesisMetricIdentifier() {
+        Arn streamArn = Arn.fromString(TEST_SPLIT.getStreamArn());
+
+        String expectedIdentifier =
+                String.format(
+                        "%s.%s.%s.%s.%s.%s.%s.%s.%s",
```

Comment:
nit: we might be better off using "." join here... This is not very readable...

Reply:
Updated.